### DISCUSSION TITLE
ngingx as reverse proxy and letsencrypt on a different PC but on the same subnet as the MC

### DISCUSSION THREAD

**Discussion by lexisman** [Ylianst/MeshCentral#5620] [Issues]
I have been scouring the forum here and I found issues similar to what I have now on my self hosted MC. I have working MC, reverse proxy and letsencrypt on one single PC, however, I have other servers running on the same subnet as the MC, so I decided to install a separate PC just for the reverse proxy, nginx and letsencrypt to proxy MC and the other servers that serves http also. 
I managed to install and run nginx and letsencrypt. However I could not properly proxy the MC. Below is my configuration file for the MC and the nginx:
nginx config:
MC version 1.1.16
server {
     listen 443 ssl;
     server_name central.xxxxxxx.xx;
     
         proxy_send_timeout 330s;
         proxy_read_timeout 330s;

     # letsencrypt by Certbot
      ssl_certificate /etc/letsencrypt/live/central.xxxxxxx.xx/fullchain.pem;
      ssl_certificate_key /etc/letsencrypt/live/central.xxxxxxx.xx/privkey.pem;
      include /etc/letsencrypt/options-ssl-nginx.conf;
      ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;

    location / {
         proxy_pass https://192.168.200.21:4430/;  - MC ip address
         proxy_http_version 1.1;

          # Allows websockets over HTTPS.
          proxy_set_header Upgrade $http_upgrade;
          proxy_set_header Connection "upgrade";
          proxy_set_header Host $host;

          # Inform MeshCentral about the real host, port and protocol
          proxy_set_header X-Forwarded-Host $host:$server_port;
          proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
          proxy_set_header X-Forwarded-Proto $scheme;
          proxy_set_header X-NginX-Proxy true;
   }
}

config.json:

 "settings": {
    "Cert": "central.xxxxxxx.xx",
    "MongoDb": "mongodb://127.0.0.1:27017/meshcentral",
    "WANonly": true,
    "_LANonly": true,
    "DbRecordsEncryptKey": "MySecretPasswordisMine",
    "sessionKey": "MyReallySecretPassword1",
    "Port": 4430,
    "AliasPort": 443,
    "RedirPort": 800,
    "agentPong": 300,
    "TlsOffload": "192.168.200.20"  - nginx ip address
  },
  "domains": {
    "": {
      "certUrl": "https://192.168.200.20:443", - nginx ip address
      "title": "Central Monitoring Systems",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true,
      "TitlePicture": "title-mycompagny.png"
    }
  }
}

Using the above configuration file failed. However, if I disable the TLSoffload, the redirection of ports, like the 443 - 4430, and the 800, then change the port setting in nginx for the MC back to 443, it does work, but the very purpose of the reverse proxy is not 100% utilized. I will appreciate any help. Thanks.

**Comment by nmincone** [comment]
Getting to this post late... did you resolve it?
I'm using NGINX RP with Let'sEncrypt SSL's in front of my MC using the URL only as cert & certURL variables.. NGINX manages the port through URL, I didn't use IP as you show above..

**Reply by lexisman** [reply]
Not really.

**Reply by lexisman** [reply]
I will try your suggestions.
